plugins/modules/load_balancer_health_monitor.py

```diff
@@ -0,0 +1,442 @@
+from ..module_utils.api import (
+    AnsibleCloudscaleBase,
+    cloudscale_argument_spec,
+)
+
+ALLOWED_STATES = ('present',
+                  'absent',
+                  )
+ALLOWED_HTTP_POST_PARAMS = ('expected_codes',
+                            'host',
+                            'method',
+                            'url_path')
+
+
+class AnsibleCloudscaleLoadBalancerHealthMonitor(AnsibleCloudscaleBase):
+
+    def __init__(self, module):
+        super(AnsibleCloudscaleLoadBalancerHealthMonitor, self).__init__(
+            module,
+            resource_name='load-balancers/health-monitors',
+            resource_key_name='pool',
+            resource_create_param_keys=[
+                'pool',
+                'timeout_s',
+                'up_threshold',
+                'down_threshold',
+                'type',
+                'http',
+                'tags',
+            ],
+            resource_update_param_keys=[
+                'delay_s',
+                'timeout_s',
+                'up_threshold',
+                'down_threshold',
+                'expected_codes',
+                'http',
+                'tags',
+            ],
+        )
+
+    def query(self):
+        # Initialize
+        self._resource_data = self.init_resource()
+
+        resource_key_pool = 'pool'
+        uuid = self._module.params[self.resource_key_uuid]
+        pool = self._module.params[resource_key_pool]
+        matching = []
+
+        # Either search by given health monitor's UUID or
+        # search the health monitor by its acossiated pool UUID (1:1)
+        if uuid is not None:
+            super().query()
+        else:
+            pool = self._module.params[resource_key_pool]
+            if pool is not None:
+
+                resources = self._get('%s' % (self.resource_name))
+
+                if resources:
+                    for health_monitor in resources:
+                        if health_monitor[resource_key_pool]['uuid'] == pool:
+                            matching.append(health_monitor)
+
+            # Fail on more than one resource with identical name
+            if len(matching) > 1:
+                self._module.fail_json(
+                    msg="More than one %s resource for pool '%s' exists." % (
+                        self.resource_name,
+                        resource_key_pool
+                    )
+                )
+            elif len(matching) == 1:
+                self._resource_data = matching[0]
+                self._resource_data['state'] = "present"
+
+        return self.pre_transform(self._resource_data)
+
+    def update(self, resource):
+        updated = False
+        for param in self.resource_update_param_keys:
+            if param == 'http':
+                for subparam in ALLOWED_HTTP_POST_PARAMS:
+                    updated = self._param_updated(subparam, resource) or updated
+            else:
+                updated = self._param_updated(param, resource) or updated
```

Comment:
We could use the parent class's update method here. This would simplify the `_param_updated` method below because it would only be relevant for the `http` params. Or am I missing something?

Reply:
updated in: https://github.com/cloudscale-ch/ansible-collection-cloudscale/pull/88/commits/8f3baabb26df6d2836a49e09df3fe922a4c1d684